cmd/default-domain/main.go

```diff
@@ -195,17 +196,25 @@ func main() {
 	if err != nil {
 		logger.Fatalw("Error finding gateway address", zap.Error(err))
 	}
+	ip := address.IP
 	if address.IP == "" {
-		logger.Info("Gateway has a domain instead of IP address -- leaving default domain config intact")
-		return
+		if address.Hostname == "" {
+			logger.Info("Gateway has neither IP nor hostname -- leaving default domain config intact")
+			return
+		}
+		ipAddr, err := net.ResolveIPAddr("ip4", address.Hostname)
```

Comment:
Looking at this with fresh eyes, are you sure that AWS guarantees that the backing IP address is sole tenancy?  I'd think the use of a CNAME was to enable them to share the IP across multiple hosts, so by doing this they'd lose important routing metadata.

Reply:
I'm not exactly sure but I guess AWS does not guarantee that the backing IP address is sole tenancy. But yes, I also guess the CNAME was to enable them to sahre the IPs as you said.

With said, the backing IP address + xip.io works fine on AWS and so I am wondering that it is good to use it for test. (I assumed that users use default-domain(xip.io) is testing for purpose.)

Reply:
Another potential reason for this is to enable them to move it to another IP, which would be problematic for this as well.